src/features/ISECompatibility.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+import * as vscode from "vscode";
+import { LanguageClient } from "vscode-languageclient";
+import { IFeature } from "../feature";
+
+/**
+ * A feature to implement commands to make code like the ISE and reset the settings.
+ */
+export class ISECompatibilityFeature implements IFeature {
+    private iseCommand: vscode.Disposable;
+    private defaultCommand: vscode.Disposable;
+    private settings = [
+        { section: "workbench.activityBar", setting: "visible", value: false },
+        { section: "debug", setting: "openDebug", value: "neverOpen" },
+        { section: "editor", setting: "tabCompletion", value: "on" },
+        { section: "powershell.integratedConsole", setting: "focusConsoleOnExecute", value: false },
+        { section: "files", setting: "defaultLanguage", value: "powershell" },
+        { section: "workbench", setting: "colorTheme", value: "PowerShell ISE" },
+    ];
+    private languageClient: LanguageClient;
+
+    constructor() {
+        this.iseCommand = vscode.commands.registerCommand("PowerShell.EnableISEMode",
+            () => this.EnableISEMode());
+        this.defaultCommand = vscode.commands.registerCommand("PowerShell.DisableISEMode",
+            () => this.DisableISEMode());
+    }
+
+    public dispose() {
+        this.iseCommand.dispose();
```

Comment:
Do these need disposing?

Reply:
Yes typically you dispose commands

Reply:
Is it an implicit aspect of the command API that a member representing a command never needs to be assigned or added to anything, it's just on registered object and matches a configuration?

Reply:
@TylerLeonhardt explained that these fields are actually the registrations of the commands. Maybe worth renaming to `iseCommandRegistration` or similar?